### PR TITLE
[Monitor] az monitor log-analytics workspace data-export: support log analytics data export

### LINKED ISSUE #13721: Add support for 'Log Analytics data export'
**Resource Provider**
microsoft.operationalinsights

**Description of Feature or Work Requested**
Feature document is available [here]( https://microsoft.sharepoint.com/:w:/t/LogAnalyticsTeam/EbhdHdsVLwBKksLCbObyVqkBSBgMZfKTzX1hs1mk2rIKLQ?e=0ytp0m):

Support these operations: 
- Create a data export rule
- Update a data export rule
   - Disable an export rule
   - Update tables for export
   - Update destination
- Get data export rule configuration
- Delete a data export rule
- Get all data export rules in a workspace

**Minimum API Version Required**
2020-03-01-preview

**Swagger Link**
- [Swaggers](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/DataExports.json)
- [rest doc](https://docs.microsoft.com/en-us/rest/api/loganalytics/dataexports) 

**Target Date**
This is already needed for private preview customers, which are using rest API currently. Once CLI is available, we can make the configuration easier for them with CLI.

### PR DESCRIPTION
**Pull request by MyronFanQiu**
**Description**  

Fix #13721 

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S172

File: src/azure-cli/azure/cli/command_modules/monitor/_params.py [hunk 1]
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):
         c.argument('retention_in_days', options_list='--retention-time', type=int, required=True)
     # endregion
 
+    # region Log Analytics Workspace Data Export
+    with self.argument_context('monitor log-analytics workspace data-export') as c:
+        c.argument('data_export_name', options_list=['--name', '-n'], help="Name of the data export rule")
+        c.argument('workspace_name', options_list='--workspace-name')
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/_params.py:387]: options_list is unnecessary

File: src/azure-cli/azure/cli/command_modules/monitor/_params.py [hunk 2]
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):
...
+        c.argument('enable_all_tables', options_list=['--all', '--enable-all-tables'], arg_type=get_three_state_flag(),
+                   help="All workspace's tables are exported when this is enabled.")
+        c.argument('table_names', nargs='+', options_list=['--tables', '-t'], help='An array of tables to export.')
+        c.argument('destination', help='The destination resource ID. It should be a storage account or an event hub namespace.')
+        c.argument('data_export_type', options_list=['--date-export-type', '--type'],
+                   help='The type of the destination resource.', arg_type=get_enum_type(['StorageAccount', 'EventHub']))
+        c.argument('event_hub_name', help='Name of an Event Hub. Not applicable when destination is Storage Account.')
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/_params.py:394]: just use destination for eventhub id and remove  event_hub_name

File: src/azure-cli/azure/cli/command_modules/monitor/_params.py [hunk 3]
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):
...
 
+    # region Log Analytics Workspace Data Export
+    with self.argument_context('monitor log-analytics workspace data-export') as c:
+        c.argument('data_export_name', options_list=['--name', '-n'], help="Name of the data export rule")
+        c.argument('workspace_name', options_list='--workspace-name')
+        c.argument('enable_all_tables', options_list=['--all', '--enable-all-tables'], arg_type=get_three_state_flag(),
+                   help="All workspace's tables are exported when this is enabled.")
+        c.argument('table_names', nargs='+', options_list=['--tables', '-t'], help='An array of tables to export.')
+        c.argument('destination', help='The destination resource ID. It should be a storage account or an event hub namespace.')
+        c.argument('data_export_type', options_list=['--date-export-type', '--type'],
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/_params.py:392]: --destination-type?

File: src/azure-cli/azure/cli/command_modules/monitor/_params.py [hunk 4]
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):
         c.argument('retention_in_days', options_list='--retention-time', type=int, required=True)
     # endregion
 
+    # region Log Analytics Workspace Data Export
+    with self.argument_context('monitor log-analytics workspace data-export') as c:
+        c.argument('data_export_name', options_list=['--name', '-n'], help="Name of the data export rule")
+        c.argument('workspace_name', options_list='--workspace-name')
+        c.argument('enable_all_tables', options_list=['--all', '--enable-all-tables'], arg_type=get_three_state_flag(),
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/_params.py:388]: --export-all-tables

File: src/azure-cli/azure/cli/command_modules/monitor/_help.py [hunk 5]
```diff
@@ -1245,6 +1245,42 @@
...
+helps['monitor log-analytics workspace data-export'] = """
+type: group
+short-summary: Manage data export ruls for log analytics workspace.
+"""
+
+helps['monitor log-analytics workspace data-export create'] = """
+type: command
+short-summary: Create a data export rule for a given workspace.
+examples:
+  - name: Create a data export rule for a given workspace.
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/_help.py:1257]: add example for exporting all tables

File: src/azure-cli/azure/cli/command_modules/monitor/_params.py [hunk 6]
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):
...
+                   help="All workspace's tables are exported when this is enabled.")
+        c.argument('table_names', nargs='+', options_list=['--tables', '-t'], help='An array of tables to export.')
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/_params.py:390]: add help "if --export-all-tables is true, this argument should not be provided."

**Pull request merged**